tracer/test/Datadog.Trace.Tests/Propagators/MultiSpanContextPropagatorTests.cs

```diff
@@ -527,6 +527,61 @@ public void ExtractAndInject_B3SingleHeader_PreserveOriginalTraceId()
             headersForInjection.Verify(h => h.Set("b3", expectedTraceParent), Times.Once());
         }
 
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void W3C_ParentExtracted_WithSameTraceIdAndParentId(bool extractFirst, bool w3CHeaderFirst)
+        {
+            // headers1 equivalent from system-tests
+            var headers = new Mock<IHeadersCollection>();
```

Comment:
tiny tiny nit: This code is all in-keeping with the existing code, so no need to change anything, but IMO we shouldn't use mock here, and instead just use one of the concrete implementations we have (or a Dictionary-based version we could add to test helpers). e.g. `NameValueHeadersCollection`